toloka_provider/sensors/toloka.py

```diff
@@ -48,3 +44,45 @@ def poke(self, context: 'Context') -> bool:
         self.log.info(f'Pool {pool_id} - {percentage}%')
 
         return pool.is_closed()
+
+
+class WaitAppBatchSensor(BaseSensorOperator):
+    """
+    Wait given App batch until it enters an of poke_on_statuses status.
+
+    :param app_project: Either an `AppProject` object or it's config or an app_project_id value.
+    :param app_project: Either an `AppBatch` object or it's config or a batch_id value.
+    :param toloka_conn_id: Airflow connection with toloka credentials.
+    :param poke_on_statuses: Container of AppBatch.Status
+    """
+
+    template_fields: Sequence[str] = ('app_project', 'batch', 'poke_on_statuses')
+
+    def __init__(
+        self,
+        *,
+        app_project: Union[AppProject, Dict, str],
+        batch: Union[AppBatch, Dict, str],
+        toloka_conn_id: str = 'toloka_default',
+        poke_on_statuses: Iterable[Union[AppBatch.Status, str]] = (AppBatch.Status.COMPLETED,),
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.app_project = app_project
+        self.batch = batch
+        self.toloka_conn_id = toloka_conn_id
+        poke_on_statuses_list = []
+        for status in poke_on_statuses:
+            poke_on_statuses_list.append(AppBatch.Status(status))
+        self.poke_on_statuses = poke_on_statuses_list
```

Comment:
`self.poke_on_statuses = [AppBatch.Status(status) for status in poke_on_statuses]`